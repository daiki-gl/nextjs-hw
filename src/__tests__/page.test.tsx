import Page from '@/app/(templates)/page'
import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { useRouter } from 'next/navigation'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn()
}));

describe('Rendering the Home page', () => {
  it('renders a buttons and checking their href attributes', () => {
        const mockPush = jest.fn();
        
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
      prefetch: jest.fn(),
      query: {},
    });
    
    render(<Page />)
    const linkElementA = screen.getByText('ボタンA');
    expect(linkElementA).toHaveAttribute('href','/search')
    
    fireEvent.click(linkElementA);
    expect(mockPush).toHaveBeenCalledWith('/search');
  })

  it("Renders Correctly", () => {
    const { asFragment } = render(<Page />);
    expect(asFragment()).toMatchSnapshot();
  });
})