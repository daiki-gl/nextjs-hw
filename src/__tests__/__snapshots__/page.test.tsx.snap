// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Rendering the Home page Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <main
      class="flex gap-[32px] row-start-2 items-center sm:items-start"
    >
      <a
        class="px-4 py-2 text-white bg-blue-500 rounded-md cursor-pointer"
        href="/search"
      >
        ボタンA
      </a>
      <a
        class="px-4 py-2 text-white bg-blue-500 rounded-md cursor-pointer"
        href="/search"
      >
        ボタンB
      </a>
    </main>
  </div>
</DocumentFragment>
`;
